### ISSUE TITLE
Improve training data and model creation. Related to validation too.

### ISSUE THREAD

**Issue by willjnz** [auspatious/ldn-lulc#43]
- Use 255 as no data value.
- Should I get training data for a region around test tiles, but not the tiles themselves? Then classify on the test tiles.
- Can the model creation be improved? Write it to S3.

2 points from Fang:
1. "Since now we have the probability output, it would be possible to check how different thresholds relate to validation accuracy and whether more filtering can be done using the probability layer. Also would need to consider how to deal with pixels not classified with high probability (should they be nodata? or labelled differently for different gap filling processes). Not sure how important/useful these are yet, but back to the PR, would it be useful to actually keep all probability layers (i.e. one per class) not just the max?"
2. Are both predict and predict_proba steps needed? Are you able to derive the predictions from the probabilities? https://github.com/auspatious/ldn-lulc/pull/36/changes/BASE..72cf63e05154e644011c4d15703c9d7703d4f2ca#r3019037183

**Comment by willjnz** [comment]
Rarotonga (089_016) 
Rarotonga training data (missing southwest part of island due to missing 2020 GeoMAD)

<img width="626" height="511" alt="Image" src="https://github.com/user-attachments/assets/42943545-9dbc-4af2-82e4-d6390bd67455" />

@alexgleith I fixed a bug that was preventing Kiribati training data creation (what I was talking about this morning). The bug was that CCI was missing data so agreement wasn't found between WC and IO.

<img width="605" height="662" alt="Image" src="https://github.com/user-attachments/assets/8585536d-d30c-4bb5-b54a-cc15852fc8c2" />

**Comment by willjnz** [comment]
Training data class distributions per tile (v0-0-2):

<img width="1036" height="820" alt="Image" src="https://github.com/user-attachments/assets/d1b1b4cf-ffe7-4658-b763-b1bdd64415d1" />

**Comment by willjnz** [comment]
@fangfy soon we will have a new version of geomad for all pacific tiles and for all years. Currently I make training data for only ~10 test tiles. For the next version of training data, what would you propose for a method? All tiles? Every `n`th tile? One tile per country?

This training data will feed into a new model. This would be a Pacific-focussed model (unless @alexgleith also wants me to run the non-Pacific tiles too).

Also should I write the training/test data split? So we could create training data for X tiles and then hold back 30% of all data for validation.

**Comment by willjnz** [comment]
Fang said:
- 1 model per region. 
- per region select a range of training tiles/countries. Use WC product to determine these based on representative land covers per class. Maybe select 5 countries per region? Definitely select a range of latitudes. Elevation as well. Fiji is more populated (speckly LC), other countries are more uniform.
- Then decide how many points per tile.
- We are aiming for: accuracy calculation. e.g. 1000 training points, 200 for validation, accuracy assesment: has a +- bounds. We want this to be e.g. 80% +- 5%.  This is directly related to how many points are used.

Pacific: volcanic, atolls, coral.

Similar: Fiji and Vanuatu. Tonga and Samoa.

517 tiles in Pacific. 300 in non-pacific.

Sample a lot. Then make a process to filter down e.g. balancing classes.

We are planning on random forest, but may change to neural network that would need ~10x points.

How to work with multiple years? Training data is just 2020 right now. We need to come up with a method for other years. e.g. if geomad has same value in year X as 2020, then that pixel's class is valid in that year. Having 1 model for all years is the goal. LS7 and LS8 are different. May need 2 times for models.

For the first version, just predict on all years with the 2020 model. This can be a baseline on which we will improve the model.

**Comment by willjnz** [comment]
@fangfy I have 15 tiles that are hopefully representative. Here are their class distributions:

<img width="1604" height="439" alt="Image" src="https://github.com/user-attachments/assets/fc36baa6-4ec8-4293-82df-aa73f0054c7a" />
